File: ios/toast_lite/Sources/toast_lite/ToastLitePlugin.swift
import Flutter
import UIKit

/// toast_lite has no native functionality — everything runs in pure Dart
/// via `Overlay`/`OverlayEntry`. This class exists only so the plugin can
/// be registered and distributed through CocoaPods/Swift Package Manager.
public class ToastLitePlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    // No method channel needed — nothing to register.
  }
}

File: android/src/main/kotlin/com/makanbangstaffnew/toast_lite/ToastLitePlugin.kt
package com.makanbangstaffnew.toast_lite

import io.flutter.embedding.engine.plugins.FlutterPlugin

/**
 * toast_lite has no native functionality — everything runs in pure Dart
 * via `Overlay`/`OverlayEntry`. This class exists only so the plugin can
 * be registered and distributed through Gradle, matching the iOS side's
 * SPM/CocoaPods dual-support setup. No method channel is needed.
 */
class ToastLitePlugin : FlutterPlugin {
    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {}

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {}
}
